### DISCUSSION TITLE
Using azd pipeline config with multiple environments or multiple projects

### DISCUSSION THREAD

**Discussion by christiannagel** [Azure/azure-dev#3585] [Q&A] [answered]
How can I use `azd pipeline config` with multiple environments and multiple projects in one repository?

It looks like no matter what environment I select; the same secrets and variables are created in GitHub. I would like to use different Azure subscriptions / resource groups / ... with different environments. I tried using GitHub environments, but secrets / variables are not stored within GitHub environments.

What about having multiple projects in one GitHub repository? How can I use `azd pipeline config` here`?

**Comment by vhvb1989** [comment, marked as answer]
Hello @christiannagel.
Up to now, `azd pipeline config` supports setting up only one azd-environment per repository. So, even if you have just one azd project locally but with more than one azd-environment, azd can only set one azd-env for CI/CD.

There's a CI/CD model which might help you, but it less common. You can ask azd to create one git remote per azd-environment (using --remote-name flag), which would get you with one gh-repository per azd-environment.  Then, depending on which remote you push your local changes is where CI/CD runs.

We have an open discussion around adding support for CI/CD environments here: https://github.com/Azure/azure-dev/issues/2373

**Reply by Lindsay-Mathieson** [reply]
> We have an open discussion around adding support for CI/CD environments here: #2373

May-2025 an theres still no solution for it.

**Reply by Lindsay-Mathieson** [reply]
> You can ask azd to create one git remote per azd-environment (using --remote-name flag), which would get you with one gh-repository per azd-environment.

Thankyou! That was the tidbit I needed to get things running.

Rather than pushing the src to Production, I am invoking the src repo workflow from Production with inherit secrets, with a little tweaking it runs exactly how I wanted. Now have workflows for auto/manual deployment to test and Manula deployment to Production.

Thank you so much. Will look into the ```azd pipeline config``` enhancements when they are ready.
